packages/next/src/server/base-server.ts

```diff
@@ -1910,7 +1911,7 @@ export default abstract class Server<
     isAppPath: boolean,
     resolvedPathname: string
   ): void {
-    const baseVaryHeader = `${RSC_HEADER}, ${NEXT_ROUTER_STATE_TREE_HEADER}, ${NEXT_ROUTER_PREFETCH_HEADER}`
+    const baseVaryHeader = `${RSC_HEADER}, ${NEXT_ROUTER_STATE_TREE_HEADER}, ${NEXT_ROUTER_PREFETCH_HEADER}, ${NEXT_ROUTER_SEGMENT_PREFETCH_HEADER}`
```

Comment:
We also need to update the vary header here I think: https://github.com/vercel/next.js/blob/5bd42716c69503e7361eba43151e85124a960056/packages/next/src/build/index.ts#L1261